Jenkinsfile

```diff
@@ -12,21 +12,30 @@ node('ios-osx') {
             '''
         }
     }
-    
+
     stage('Prepare') {
-        
+
         sh '''#!/bin/bash -l -x
             rvm use ruby-2.4.0
             gem install xcpretty -N
             gem install cocoapods
             brew update
             brew install xctool
             brew install carthage
+            brew install yarn
             pod --version
-            echo A |./bootstrap.sh 
+        '''
+        // load/restore carthage build directory
+        sh '''#!/bin/bash -l -x
+            CART_CACHE=/tmp/carthage_cache_`md5 -q Cartfile`.tar; tar -xf $CART_CACHE ; echo A |./bootstrap.sh && tar -cf $CART_CACHE Carthage Cartfile.resolved
```

Comment:
would be nicer to have groovy helper like `withCarthageCache` or something like that